sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java

```diff
@@ -82,7 +105,9 @@ public final class OpenAIClient {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<BinaryData> getEmbeddingsWithResponse(
             String deploymentId, BinaryData embeddingsOptions, RequestOptions requestOptions) {
-        return this.serviceClient.getEmbeddingsWithResponse(deploymentId, embeddingsOptions, requestOptions);
+        return openAIServiceClient != null
+            ? openAIServiceClient.getEmbeddingsWithResponse(deploymentId, embeddingsOptions, requestOptions)
+            : azureServiceClient.getEmbeddingsWithResponse(deploymentId, embeddingsOptions, requestOptions);
```

Comment:
If we are using ternary operator, we should do it in both sync and async clients to be consistent.